python/ray/serve/tests/test_deploy.py

```diff
@@ -614,13 +613,19 @@ def call():
         else:
             ret = requests.get(f"http://localhost:8000/{name}").text
 
-        return ret.split("|")[0], ret.split("|")[1]
+        rets = ret.split("|")
+        if len(rets) == 2:
+            return rets[0], rets[1]
+        elif len(rets) == 1:
+            return rets[0], "<NULL>"
+        else:
+            return "<NULL>", "<NULL>"
```

Comment:
An `IndexError` was being raised in one of the processes executing this task. I added this to handle the corner cases. I can create a utility function  to do this and replace `return ret.split("|")[0], ret.split("|")[1]` with a call to that function. Let me know. cc: @pcmoritz @simon-mo

Reply:
Do you happen to remember what value `ret` was? Was the `IndexError` caused because `ret.split("|")[1]` failed?

Reply:
> I want to make sure that the corner case wasn't actually an issue with the test's correctness.

Yes. I will re-run the tests tomorrow and post the logs here.

Reply:
```
  File "python\ray\_raylet.pyx", line 640, in ray._raylet.execute_task
    outputs = function_executor(*args, **kwargs)
  File "C:\Users\gagan\ray_project\ray\python\ray\serve\tests\test_deploy.py", line 675, in call
    return ret.split("|")[0], ret.split("|")[1]
IndexError: list index out of range
 pid=2196) 2022-01-06 08:14:44,896      INFO deployment_state.py:939 -- Removing 4 replicas from deployment 'test'. component=serve deployment=test
 pid=2588) ret:  Path '/test' not found. Please ping http://.../-/routes for route table.
 pid=7964) ret:  2|564
 pid=3992) ret:  Path '/test' not found. Please ping http://.../-/routes for route table.
 pid=4020) ret:  Path '/test' not found. Please ping http://.../-/routes for route table.
 pid=11184) ret:  2|7256
 pid=10512) ret:  Path '/test' not found. Please ping http://.../-/routes for route table.
 pid=4988) ret:  Path '/test' not found. Please ping http://.../-/routes for route table.
 pid=7044) ret:  2|7076
 pid=6088) ret:  2|9404
 pid=7968) ret:  2|9404
 ray\serve\tests\test_deploy.py::test_redeploy_scale_up[False]
```

I think `Path '/test' not found. Please ping http://.../-/routes for route table.` means that deployment didn't complete when `requests.get` got executed and accessed `"http://localhost:8000/test"`? I think I should also return timestamps of returns for debugging purposes. That might help in figuring out when the `v1` and `v2` objects were deployed and when actually the `requests.get` accessed  `"http://localhost:8000/test"` the location. Though we also need to see if `Path '/test' not found. Please ping http://.../-/routes for route table.` is anyways affecting the test.

P.S. Result of another race condition may be?